cli/backend_base.py

```diff
@@ -207,14 +208,65 @@ def _ship_certs(self, cluster, saltmaster_ip):
 
         return platform_certs_tarball
 
+    def _fill_volume_config(self, volume_dict):
+        ''' returns the generated volumes for kafka and datanodes '''
+        data_volume_list = []
+        if self._pnda_env['disk_volumes']['GENERATE_VOLUME_CONFIG'] == "YES":
```

Comment:
This check should be done in _check_config if GENERATE_VOLUME_CONFIG then DATANODE_VOLUME_COUNT must be > 0. This will prevent someone waiting until this bootstrap phase to discover the config is wrong. Also consider defaulting DATANODE_VOLUME_COUNT to 1 so that the default config works when GENERATE_VOLUME_CONFIG is turned on.